src/cargo/core/registry.rs

```diff
@@ -166,9 +166,9 @@ impl<'cfg> PackageRegistry<'cfg> {
 
     pub fn register_lock(&mut self, id: PackageId, deps: Vec<PackageId>) {
         let sub_map = self.locked.entry(id.source_id().clone())
-                                 .or_insert(HashMap::new());
+                                 .or_insert_with(HashMap::new);
```

Comment:
According to https://github.com/Manishearth/rust-clippy/wiki#or_fun_call, the old version will indeed always be called.